File: backend/app/routers/riaas/chapters.py
from __future__ import annotations

from fastapi import APIRouter, Depends

from app.deps import require_riaas_access

router = APIRouter(
    prefix="/api/riaas",
    tags=["riaas"],
    dependencies=[Depends(require_riaas_access)],
)

CHAPTERS = [
    {"slug": "gtm-overview", "title": "GTM Efficiency Overview"},
    {"slug": "win-loss", "title": "Win/Loss & Benchmark"},
    {"slug": "pipeline-health", "title": "Pipeline Health Assessment"},
    {"slug": "coach", "title": "Coach (People Insights)"},
    {"slug": "gtm-process", "title": "GTM Process Optimisation"},
]


@router.get("/chapters")
def list_chapters() -> list[dict]:
    return CHAPTERS

File: frontend/src/lib/chapters.ts
export interface ChapterDef {
  slug: string;
  title: string;
  navLabel: string;
}

export const CHAPTERS: ChapterDef[] = [
  { slug: "gtm-overview", title: "GTM Efficiency Overview", navLabel: "GTM Overview" },
  { slug: "win-loss", title: "Win/Loss & Benchmark", navLabel: "Win/Loss" },
  { slug: "pipeline-health", title: "Pipeline Health Assessment", navLabel: "Pipeline Health" },
  { slug: "coach", title: "Coach (People Insights)", navLabel: "Coach" },
  { slug: "gtm-process", title: "GTM Process Optimisation", navLabel: "GTM Process" },
];

export function chapterBySlug(slug: string): ChapterDef | undefined {
  return CHAPTERS.find((c) => c.slug === slug);
}
